### DISCUSSION TITLE
Running Teleport v13 and try an update to v14, 15 and 16

### DISCUSSION THREAD

**Discussion by techsupportkoeln** [gravitational/teleport#43583] [Q&A] [answered]
My self hosted teleport server is currently still running on version 13 on an Ubuntu 22.04 LTS. The goal would be to update Teleport to version 16.

I understood how I can update the server version step by step using the following instructions: https://goteleport.com/docs/ver/14.x/upgrading/self-hosted-linux/

However, updating the agents is causing me some difficulties.

If I understand these [instructions ](https://goteleport.com/docs/ver/14.x/upgrading/self-hosted-automatic-agent-updates/)correctly, I absolutely need Amazon S3 or Google Cloud Storage for the automatic agent updates. Is that correct? I find that strange.

If I use the following commands on the client for an manual teleport agent update, I can update up to version 14. From version 15 oder 16 I get an error message on my Ubuntu 22.04 LTS clients.

Anyone with an idea how to fix this?

```
root@XXXXX:~# sudo curl https://apt.releases.teleport.dev/gpg \
-o /usr/share/keyrings/teleport-archive-keyring.asc
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100  3163  100  3163    0     0  32839      0 --:--:-- --:--:-- --:--:-- 32947
root@XXXXX:~# source /etc/os-release
root@XXXXX:~# echo "deb [signed-by=/usr/share/keyrings/teleport-archive-keyring.asc] \
https://apt.releases.teleport.dev/${ID?} ${VERSION_CODENAME?} stable/v16" \
| sudo tee /etc/apt/sources.list.d/teleport.list > /dev/null
root@XXXXX:~# sudo apt-get update
Hit:1 https://apt.releases.teleport.dev/ubuntu jammy InRelease
Hit:2 http://archive.ubuntu.com/ubuntu jammy InRelease                               
Hit:3 http://archive.ubuntu.com/ubuntu jammy-updates InRelease                       
Hit:4 http://archive.ubuntu.com/ubuntu jammy-security InRelease                               
Get:5 https://apt.releases.teleport.dev/ubuntu jammy/stable/v16 amd64 Packages [2636 B]       
Hit:6 https://repo.zabbix.com/zabbix/6.4/ubuntu jammy InRelease             
Fetched 2636 B in 1s (3636 B/s)                  
Reading package lists... Done
root@XXXXX:~# sudo apt-get install teleport-ent-updater
Reading package lists... Done
Building dependency tree... Done
Reading state information... Done
The following additional packages will be installed:
  teleport-ent
The following NEW packages will be installed:
  teleport-ent teleport-ent-updater
0 upgraded, 2 newly installed, 0 to remove and 4 not upgraded.
Need to get 188 MB of archives.
After this operation, 759 MB of additional disk space will be used.
Do you want to continue? [Y/n] y
Get:1 https://apt.releases.teleport.dev/ubuntu jammy/stable/v16 amd64 teleport-ent amd64 16.0.1 [188 MB]
Get:2 https://apt.releases.teleport.dev/ubuntu jammy/stable/v16 amd64 teleport-ent-updater amd64 16.0.1 [9484 B]
Fetched 188 MB in 2s (112 MB/s)           
Selecting previously unselected package teleport-ent.
(Reading database ... 21956 files and directories currently installed.)
Preparing to unpack .../teleport-ent_16.0.1_amd64.deb ...
Unpacking teleport-ent (16.0.1) ...
dpkg: error processing archive /var/cache/apt/archives/teleport-ent_16.0.1_amd64.deb (--unpack):
 trying to overwrite '/lib/systemd/system/teleport.service', which is also in package teleport 13.4.26
dpkg-deb: error: paste subprocess was killed by signal (Broken pipe)
Selecting previously unselected package teleport-ent-updater.
Preparing to unpack .../teleport-ent-updater_16.0.1_amd64.deb ...
Unpacking teleport-ent-updater (16.0.1) ...
Errors were encountered while processing:
 /var/cache/apt/archives/teleport-ent_16.0.1_amd64.deb
E: Sub-process /usr/bin/dpkg returned an error code (1)
root@XXXXX:~# 
```

**Comment by webvictim** [comment, marked as answer]
With a Teleport v16 server, you don't need S3 or GCS for self-hosted automatic upgrades - assuming that your URL is `teleport.example.com`, Teleport will serve its own version on https://teleport.example.com/v1/webapi/automaticupgrades/channel/default/version

Bear in mind that you do, however, need to be using Teleport Enterprise rather than Teleport OSS, as the Teleport updater package only installs and manages the `teleport-ent` package.

I can see from your logs that you're trying to install `teleport-ent` over `teleport`, which is causing a conflict. `sudo apt-get -y install teleport` should upgrade you to Teleport OSS v16.0.1.

If you have a license to use Teleport Enterprise, you should remove the `teleport` package (with `sudo apt-get remove teleport`) and then install `teleport-ent-updater` again (`sudo apt-get install teleport-ent-updater`) which will also install `teleport-ent` as a dependency.
